shared/actions/fs/index.js

```diff
@@ -54,52 +54,58 @@ const getSyncConfigFromRPC = (tlfName, tlfType, config: RPCTypes.FolderSyncConfi
   }
 }
 
-const tlfListToGetSyncConfigPromise = (state, tlfType) =>
-  Array.from(state.fs.tlfs.get(tlfType, I.Map()).keys()).map(tlfName =>
-    RPCTypes.SimpleFSSimpleFSFolderSyncConfigAndStatusRpcPromise({
-      path: Constants.pathToRPCPath(Constants.tlfTypeAndNameToPath(tlfType, tlfName)),
-    }).then(result => ({
-      syncConfig: getSyncConfigFromRPC(tlfName, tlfType, result.config),
-      tlfName,
-      tlfType,
-    }))
-  )
+const rpcFolderToTlfNameAndType = (folder: RPCTypes.Folder) => {
+  switch (folder.folderType) {
+    case RPCTypes.favoriteFolderType.private:
+      return {tlfName: folder.name, tlfType: 'private'}
+    case RPCTypes.favoriteFolderType.public:
+      return {tlfName: folder.name, tlfType: 'public'}
+    case RPCTypes.favoriteFolderType.team:
+      return {tlfName: folder.name, tlfType: 'team'}
+    default:
+      return null
+  }
+}
 
-// TODO (KBFS-4047): make a SimpleFS RPC for this case where we are asking for
-// all.
 const loadSyncConfigForAllTlfs = (state, action) =>
-  Promise.all(
-    // TODO: sometimes we get an error if a TLF is not backed by team.
-    [/* 'private', 'public', */ 'team']
-      .map(tlfType => tlfListToGetSyncConfigPromise(state, tlfType))
-      // $FlowIssue hasn't learnt about .flat yet.
-      .flat()
-  ).then(results => {
-    const payloadMutable = results.reduce(
-      (payload, {syncConfig, tlfType, tlfName}) => ({
-        ...payload,
-        [tlfType]: payload[tlfType].set(tlfName, syncConfig),
-      }),
+  RPCTypes.SimpleFSSimpleFSSyncConfigAndStatusRpcPromise().then(({folders}) => {
+    if (!folders) {
+      return null
+    }
+    const mutable = folders.reduce(
+      (mutable, {folder, config}) => {
+        const tlfNameAndType = rpcFolderToTlfNameAndType(folder)
+        return tlfNameAndType
+          ? {
+              ...mutable,
+              [tlfNameAndType.tlfType]: mutable[tlfNameAndType.tlfType].set(
+                tlfNameAndType.tlfName,
+                getSyncConfigFromRPC(tlfNameAndType.tlfName, tlfNameAndType.tlfType, config)
+              ),
+            }
+          : mutable
+      },
       {
         private: I.Map().asMutable(),
         public: I.Map().asMutable(),
         team: I.Map().asMutable(),
       }
     )
     return FsGen.createTlfSyncConfigsLoaded({
-      private: payloadMutable.private.asImmutable(),
-      public: payloadMutable.public.asImmutable(),
-      team: payloadMutable.team.asImmutable(),
+      private: mutable.private.asImmutable(),
```

Comment:
why the rename? I thought the old name was more descriptive.

Reply:
Oh; I rewrote this function and didn't consciously rename it. Will change it back.